presto-main/src/main/java/com/facebook/presto/type/DoubleOperators.java

```diff
@@ -217,6 +217,9 @@ public static long castToTinyint(@SqlType(StandardTypes.DOUBLE) double value)
     @SqlType(StandardTypes.BIGINT)
     public static long castToLong(@SqlType(StandardTypes.DOUBLE) double value)
     {
+        if (value >= Long.MAX_VALUE || value <= Long.MIN_VALUE) {
```

Comment:
Bug is here. You should have = equal.

Think about it and understand why this bug only affected negative test case, but not the positive test case.